homeassistant/components/comfoconnect/fan.py

```diff
@@ -43,24 +43,34 @@ def __init__(self, name, ccb: ComfoConnectBridge) -> None:
 
     async def async_added_to_hass(self):
         """Register for sensor updates."""
+        _LOGGER.debug("Registering for fan speed")
+        async_dispatcher_connect(
+            self.hass,
+            SIGNAL_COMFOCONNECT_UPDATE_RECEIVED.format(SENSOR_FAN_SPEED_MODE),
+            self._handle_update,
+        )
         await self.hass.async_add_executor_job(
             self._ccb.comfoconnect.register_sensor, SENSOR_FAN_SPEED_MODE
         )
-        async_dispatcher_connect(
-            self.hass, SIGNAL_COMFOCONNECT_UPDATE_RECEIVED, self._handle_update
-        )
 
-    def _handle_update(self, var):
+    def _handle_update(self, value):
         """Handle update callbacks."""
-        if var == SENSOR_FAN_SPEED_MODE:
-            _LOGGER.debug("Received update for %s", var)
-            self.schedule_update_ha_state()
+        _LOGGER.debug(
+            "Handle update for fan speed (%d): %s", SENSOR_FAN_SPEED_MODE, value
+        )
+        self._ccb.data[SENSOR_FAN_SPEED_MODE] = value
+        self.schedule_update_ha_state()
 
     @property
     def should_poll(self) -> bool:
         """Do not poll."""
         return False
 
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return f"{self._ccb.unique_id}-fan"
```

Comment:
The entity registry is aware of both the integration domain and the platform domain. We can remove the fan suffix.

Reply:
This was done for consistency with the sensor plaform, where I append the sensor index.
https://github.com/home-assistant/home-assistant/pull/28817/files#diff-e4143e4a20a8ad2a277a56b3d5de7342R270-R272

Actually, I could also append `SENSOR_FAN_SPEED_MODE`, since that's actually how it works in the sensors.

I haven't looked into the way to define devices in hass (with the new manage devices section), but I thought if this also required a unique id, `{self._ccb.unique_id}` without a suffix would be more appropriate for that, hence the suffix.

Reply:
If there's only one entity per unique_id and platform we don't need a suffix. Please remove it.

Reply:
Done!